pkg/kubectl/cmd/util/shortcut_restmapper.go

```diff
@@ -70,6 +68,14 @@ func (e ShortcutExpander) getAll() []schema.GroupResource {
 	return availableAll
 }
 
+func (e ShortcutExpander) getServerResources() ([]*metav1.APIResourceList, error) {
+	if e.discoveryClient == nil {
```

Comment:
Fail during construction if there's no `discoveryClient`, then just let this panic.  Also, is there a good reason to expose the `ShortcutExpander` struct?  That would make it slightly cleaner to fail.